tools/PI/DevHome.PI/Controls/AddToolControl.xaml.cs

```diff
@@ -101,55 +161,290 @@ private void ClearValues()
-            argType,
-            PrefixTextBox.Text ?? string.Empty,
-            OtherArgsTextBox.Text ?? string.Empty,
+            activationType,
+            ArgumentsTextBox.Text,
+            selectedApp?.AppUserModelId ?? string.Empty,
+            selectedApp?.IconFilePath ?? string.Empty,
             IsPinnedToggleSwitch.IsOn);
     }
 
-    private void UpdateSampleCommandline(object sender, TextChangedEventArgs e)
+    private void CancelButton_Click(object sender, RoutedEventArgs e)
     {
-        UpdateSampleCommandline();
+        ClearValues();
     }
 
-    private void UpdateSampleCommandline(object sender, RoutedEventArgs e)
+    private async void AppListButton_Click(object sender, RoutedEventArgs e)
     {
-        UpdateSampleCommandline();
+        allApps.Clear();
+        SortedApps.Clear();
+
+        itemCount = GetShortcuts();
+        if (itemCount == 0)
+        {
+            Log.Error("Error getting shortcuts");
+        }
+
+        var packageCount = GetPackages();
+        if (packageCount == 0)
+        {
+            Log.Error("Error getting packages");
+        }
+
+        itemCount += packageCount;
+        if (itemCount == 0)
+        {
+            Log.Error("Error getting list of installed apps");
+            return;
+        }
+
+        // We get most of the data on a background thread, which
+        // reports intermittent progress.
+        IsLoading = true;
+        var progress = new Progress<int>(percent =>
+        {
+            // Update the progress report.
+            LoadingProgressTextRing.Value = percent;
+        });
+
+        await ProcessItemsAsync(progress);
+
+        foreach (var app in allApps)
+        {
+            SortedApps.Add(app);
+        }
+
+        IsLoading = false;
+    }
+
+    private int GetShortcuts()
+    {
+        var count = 0;
+
+        // Search for .lnk files in the per-user and all-users Start Menu Programs directories.
+        // %APPDATA%\Microsoft\Windows\Start Menu\Programs
+        var startMenuProgramsPath =
+            Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.StartMenu), "Programs");
+
+        // %ProgramData%\Microsoft\Windows\Start Menu\Programs
+        var commonStartMenuProgramsPath =
+            Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.CommonStartMenu), "Programs");
+
+        shortcuts = [];
+        if (Directory.Exists(startMenuProgramsPath))
+        {
+            shortcuts.AddRange(Directory.GetFiles(startMenuProgramsPath, "*.lnk", SearchOption.AllDirectories));
+        }
+
+        if (Directory.Exists(commonStartMenuProgramsPath))
+        {
+            shortcuts.AddRange(Directory.GetFiles(commonStartMenuProgramsPath, "*.lnk", SearchOption.AllDirectories));
+        }
+
+        if (shortcuts is not null)
+        {
+            count = shortcuts.Count;
+        }
+
+        return count;
     }
 
-    private void UpdateSampleCommandline()
+    private int GetPackages()
```

Comment:
NIT: change name to `CountPackages`  A "Count" method returning an int makes sense.  "Get" sounds like this should return a list of packages.

Reply:
The main purpose of the method is to get the packages. Returning the count is a secondary requirement.